src/app/components/core/configuration-manager/view-config-item/view-config-item.component.ts

```diff
@@ -352,4 +342,65 @@ export class ViewConfigItemComponent implements OnInit, OnChanges, AfterViewChec
     }
   }
 
+  checkValidity(item: any = '', value = '') {
+    if (!isEmpty(this.categoryConfigurationData)) {
+      const data = this.categoryConfigurationData.value[0];
+      const config = [];
+      for (const k in data) {
+        if (data.hasOwnProperty(k)) {
+          data[k].value = data[k].value !== undefined ? data[k].value : data[k].default;
+          if (item !== '' && k === item.key) {
+            data[k].value = value;
+          }
+          config.push(
+            {
+              key: k,
+              value: data[k].value
+            });
+        }
+      }
+
+      for (const k in data) {
+        if (data.hasOwnProperty(k)) {
+          data[k].key = k;
+          if (data[k].hasOwnProperty('validity')) {
+            data[k].validityExpression = data[k].validity;
+            config.forEach(element => {
+              data[k].validityExpression = data[k].validityExpression.includes(element.key) ? data[k].validityExpression
+                .replace(new RegExp(element.key, 'g'), `'${element.value}'`) : data[k].validityExpression;
+            });
+          }
+        }
+      }
+
+      for (const k in data) {
+        if (data.hasOwnProperty(k)) {
+          if (data[k].hasOwnProperty('validity')) {
+            if (data[k]['validity'].trim() !== '') {
+              try {
+                // tslint:disable-next-line: no-eval
+                const e = eval(data[k].validityExpression);
+                if (typeof (e) !== 'boolean') {
+                  console.log('Validity expression', data[k].validityExpression, 'for', k, 'evlauted to non-boolean value ', e);
```

Comment:
make `data[k].validityExpression` -> `data[k].validity` to log origin expression 

typo `evlauted` -> `evaluated`

also remove the space after `... non-boolean value ` and before e value